AnkiDroid/src/main/java/com/ichi2/anki/BackupManager.kt

```diff
@@ -122,37 +110,52 @@ open class BackupManager {
         return true
     }
 
-    fun isBackupUnnecessary(colFile: File, deckBackups: Array<File?>): Boolean {
-        val len = deckBackups.size
+    fun isBackupUnnecessary(colFile: File, backups: Array<File>): Boolean {
+        val len = backups.size
 
         // If have no backups, then a backup is necessary
         return if (len <= 0) {
             false
-        } else deckBackups[len - 1]!!.lastModified() == colFile.lastModified()
+        } else backups[len - 1].lastModified() == colFile.lastModified()
 
         // no collection changes means we don't need a backup
     }
 
-    /** Returns the last date of a backup, or null if none  */
-    fun getLastBackupDate(deckBackups: Array<File?>, df: SimpleDateFormat): Date? {
-        // TODO: It appears that we can just use a loop here
-        // TODO: This doesn't seem to work properly - we don't use a min()
-        var len = deckBackups.size
-        var lastBackupDate: Date? = null
-        while (lastBackupDate == null && len > 0) {
-            lastBackupDate = try {
-                len--
-                df.parse(
-                    deckBackups[len]!!.name.replace(
-                        "^.*-(\\d{4}-\\d{2}-\\d{2}-\\d{2}-\\d{2}).colpkg$".toRegex(), "$1"
-                    )
-                )
+    /**
+     * @return filename with pattern collection-yyyy-MM-dd-HH-mm based on given time parameter
+     */
+    fun getNewBackupName(time: Time): String? {
+        /** Changes in the file name pattern should be updated as well in
+         * [getBackupTimeStrings] and [com.ichi2.anki.dialogs.DatabaseErrorDialog.onCreateDialog] */
+        val cal: Calendar = time.gregorianCalendar()
+        val backupFilename: String = try {
+            String.format(Utils.ENGLISH_LOCALE, "collection-%s.colpkg", df.format(cal.time))
+        } catch (e: UnknownFormatConversionException) {
+            Timber.e(e, "performBackup: error on creating backup filename")
```

Comment:
```suggestion
            Timber.w(e, "performBackup: error on creating backup filename")
```

You seem to be recovering from the error with a return false above, we try to use error for things that will likely require an app restart - something basically unrecoverable. Warning seems about right as a log level here